docs/the-git-based-process-you-need-for-mainframe-development-DRAFT/the-git-based-development-process-for-mainframe-development.md

```diff
@@ -35,14 +39,49 @@ As Git became the de-facto version control system in today's IT world, new termi
 Many mainframe development teams follow a release-based or iteration-based process to deliver incremental updates to a pre-defined production runtime.
 
 ## Characteristics of mainline-based development with feature branches 
+### Starting simple
 
 The mainline-based development approach[^3] with *feature* branches is a simple and structured workflow to implement, integrate, and deliver changes with an early integration process flow using a single long-living branch: *main*. Developers work in isolation in *feature* branches to implement changes to the source code, and ideally test the changes in a specific environment.
 
-Additional *epic*[^4] and *release maintenance* branches accommodate specific development workflows and allow the model to scale. The latter two branches exist for the duration of the epic or release maintenance and are short living branches.
-
 This approach can be compared to a trunk-based branching model that leverages feature branches. A similar workflow like outlined in this publication is also documented by Microsoft without giving it a name[^5].
 
-Developers use feature branches to work on any type of change, including defect fixes. In the typical development workflow of this branching strategy, the changes are merged into the *main* branch. The implemented changes of the iteration are then delivered collectively as part of the next release. Each development team decides how long an iteration is: we advocate for working towards smaller, quicker release cycles, but this model can also be used with longer iterations. Due to business or technical reasons, the merging of features into the *main* branch can also be delayed. Although being a discouraged practice, the recommendation is to group such features in a specific epic branch, as described later.
+The `main` branch is the point of reference for the entire history of the mainline changes to the code base. `main` should be a *protected* 
+branch - **all** changes originate on a branch created to hold them until they are ready to be merged. Using a branch-and-merge approach is
+natural in a git-based SCM and very light-weight both in terms of resource consumption and developer experience.
+
+#### All changes start on a dedicated branch
```

Comment:
their use of the term 'topic branch' is entirely in line with the intent here so we could use the term so long as we introduce it correctly - I was trying to really simplify the language with the minimum of set of terms to be familiar with. I think in the git-scm book it works well as it's really a direct compare with the 'long-running branches' they introduce just before.

Reply:
Re-reading how I'd worded this paragraph oriented around *feature* branches, I can see how that term is too specific - I would think that, broadly speaking, we'd actually end up with two types of *topic* branches:
- *feature* branches - branches for, well, features (being net-new function)
- *fix* or *bug-fix*(?) branches - branches for fixing defects/issues 

Do we want to go the route of two top-level types of branch - *long-lived* and *topic*, and then two sub-types of *topic* - *feature* and *bug-fix*? And then when we scale up, there's a third sub-type of *topic* branch - *epic*.

I would further argue that *release* branches then become another top level type of branch since they are not *long-lived* since they should not live as long as `main` (or `develop` if we bothered to have one), but also they are never intended to merge any where like *topic* branches do.

So we'd end up with 3 top-level types and 2 sub-types of one of those types:
- *long-lived* - eg `main`
- *topic*
   - *feature* (could provide an example of naming feature branches)
   - *bug-fix* (ditto)
- *release*

Reply:
I've taken a lightweight approach in the latest commit and merely introduced the term *topic branch* by saying that the term is used in the git-scm book and that we regard feature branches as a sub-type. Otherwise I think we'd end up with a third term and *topic* would really only be that 'abstract super-type' of *feature* and *bug-fix*.